### DISCUSSION TITLE
Environments not available on my organization

### DISCUSSION THREAD

**Discussion by franek** [community/community#26262] [Enterprise]
<p>Hi,</p>
<p>I am very excited about <a href="https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions: Environments, environment protection rules and environment secrets (beta) - GitHub Changelog</a></p>
<blockquote>
<p>In this open beta there is no need to sign up, all existing GitHub organizations and accounts can use the new capabilities in their public repositories and GitHub Enterprise Cloud organizations can use them in all repositories.</p>
</blockquote>
<p>It is not available into my organisation repositories only in my private repositories. Would it possible to enable it on my organization ? If yes, how ?</p>
<p>Thanks,</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>At this time the private repository use is only available on GitHub Enterprise Cloud so you would have to upgrade your organisation to that plan.</p>
<p>We have detailed feature comparisons at the bottom of <a href="https://github.com/pricing">our pricing page</a>.</p>

**Comment by franek** [comment]
<p>Thanks for your answer.<br>
Do you plan to provide this feature to Github Team Plan ?<br>
Thanks,<br>
François</p>

**Comment by canuckjacq** [comment]
<p>We do refresh our offerings from time to time, but right now I don’t know.</p>

**Comment by jeffpapp** [comment]
<p>I’d also be interested in using the environments feature in our Team plan.  It seems like a pretty steep jump from $4/user/month to $21/user/month just to be able to manage deployments to a staging and production environment.</p>
<p>I could totally see the environment protection rules being part of the enterprise plan but just having environments seems like a feature most small teams would want to use</p>

**Comment by Steph0** [comment]
<p>Same here. We can’t afford at the moment to jump to $21/user/month.<br>
I had to create a tool to have a centralized “environments” configurations repository for our Github Actions: <a href="https://github.com/Steph0/dotenv-configserver" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Steph0/dotenv-configserver: Load dotenv files to GITHUB_ENV from remote repository</a></p>
<p>It works like a Spring Cloud config server<br>
Feel free to use it if it can fill the gap until Teams plan get the feature maybe.</p>
<p>It does not provide the features of env protections rules but, it provides at least that possibility to go over 10 inputs, trigger generic actions/workflow based on an environment variables, etc.<br>
Used in pair with carefully crafted restrictions on push/pull, PRs, checks, etc… it does the job for us!</p>

**Comment by fredrbl** [comment]
<p>Same here.<br>
I have talked to soo many that’s in the same boat as us - not having environment in the teams plan is a real party pooper, and we are too small to go up to enterprise plan.<br>
Pleeeease Github, make it accessible for private repos in teams plan. Or else we have to go back to gitlab again quite soon. (unless you make it possible to manually approve steps in workflows on other ways)…</p>

**Comment by abjrcode** [comment]
<p>Hi GitHub support,</p>
<p>Just wanting to put some pressure on you guys, nothing but love. I really really want this feature in the Teams plan as our company is a small startup that cannot yet afford the Enterprise plan.</p>
<p>All the best</p>

**Comment by nick-zh** [comment]
<p><a class="mention" href="/canuckjacq">@canuckjacq</a> any update if this might come to the teams plan anytime soon?<br>
Thanks in advance for an update :pray:<br>
This is the one feature i am missing here that i have on CircleCI (to approve steps)</p>

**Comment by moraisp** [comment]
<p>Work in a Brazilian org and there’s no way we can afford the 21usd per user in the Enterprise plan. This missing feature would make a world of a difference to us.</p>

**Comment by iamdev42** [comment]
<p>What a bummer. We will be trying now what <a class="mention" href="/steph0">@Steph0</a> suggested and see how it works… but this definitely should be considered for free version (or at least Dev one).<br>
Hope workaround works, otherwise we need to check what is out there…</p>

**Comment by nalejandroveron** [comment]
<p>I come here to put a +1. That feature is great and it would help us to properly require approvals to deploy. Sadly we’re a small bootstrapped team so the jump in pricing is too big for us.</p>

**Comment by dezren39** [comment]
<p>Locking security features behind a paywall is an antipattern, you should really reconsider-- especially considering actions is becoming a larger and larger part of the ci cd ecosystem. Branch protection and environments need to be unlocked.</p>

**Comment by JCMais** [comment]
<p>Are there plans to add support for this on the Team plan? The $21/user/month of the enterprise is a no-go for us</p>

**Comment by lluppes** [comment]
<p>This definitely should be added to the Teams plan, not strictly limited to the Enterprise. It’s a shame to limit this feature which everyone should be using to just Enterprise.</p>

**Comment by cemo** [comment]
<p>I think it was announced like beta so it is part of enterprise plan for limiting users.<br>
This is definitely not an enterprise feature something like Single Sign On or other enterprise features. Please bring to Teams plan.</p>

**Comment by hanscg** [comment]
<p>+1 on adding it to the Team plan.<br>
We’re moving on from CircleCI to Github Actions but the lack of Environments are keeping us from fully moving all of our workflows.</p>

**Comment by Marche86** [comment]
<p>+1 on adding it to the Team plan.</p>

**Comment by Lincon-Freitas** [comment]
<p>+1 need this feature.</p>

**Comment by superhp** [comment]
<p>+1 regarding adding the environments management for at least Team plan. It is pitty we started with GitHub and now we have to consider alternatives because of this. It would help to know the roadmap if it is even on the table or the only solution is Enterprice plan.</p>

**Comment by fbuinosquy1985** [comment]
need this.

**Comment by Kowagirl** [comment]
It looks like this was recently added to the Team plan 😃

**Comment by nick-zh** [comment]
@Kowaman you are absolutely right :tada: , we can see that on the updated [env page](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#about-environments), seems thought that some features like reviewers / manual approval are still restricted to enterprise :cry:

**Reply by GoranHalvarsson** [reply]
That sucks

**Comment by aarvi13** [comment]
In environment approval for enterprise account, reviewers option not showing.

**Comment by fbuinosquy1985** [comment]
need approval on team plan .

**Reply by Amaytavanandi** [reply]
In the absence of this how are people managing their production releases? 

We need a pipeline that deploys to UAT and then Production after a review and approval. What are people doing to prevent builds going straight to production following a UAT release?

The only way would be to replicate the CI/CD for each environment and kick them off manually?

**Reply by Piedone** [reply]
We have a separate repository with a different set of user accesses for what you call a production release. The necessary secrets are also only available there.

You can't mimic Environments with branch protection rules, for example, because all people can see all branches in a given repo. Thus, anyone could take the prod release workflow and run in another branch.

**Comment by akostadinov** [comment]
Github, could you clarify this feature in the docs [1]?

It reads:
> Environments, environment secrets, and environment protection rules are available in public repositories for all products.

But I don't see environments anywhere neither in org settings, nor in public repo settings. We seem to be on a legacy plan and most our repos are public and open source. So according to the above description, environments should be available in those. 

[1] https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

**Comment by Thorsten-Totemic** [comment]
Github team, please provide some clarifications. This was a requirement for us to move from circleCI to Github actions and according to the docs it should work as we are on the GH teams plan. So I started the migration and almost finished until I find out it doesn't work although it says so in the docs:
```
Note: Deployment protection rules are only available in public repositories if you are using GitHub Free, GitHub Pro, or GitHub Team. For access to deployment protection rules in private or internal repositories, you must use GitHub Pro, GitHub Team, or GitHub Enterprise. For more information on switching your subscription...
```
Please advise as this is a really important feature

**Comment by dennmee** [comment]
Please give an update on this as the docs are misleading. I was configuring this thinking i found a solution for manual approvals. only to find out the deployment rules options are not present in Team.
Either fix/clarify the docs or just add the feature (that should have been here in the first place 36 years ago)

**Comment by cin-yong** [comment]
Please add this feature to github team plan, our org want this so bad !

**Comment by Spixmacaw** [comment]
Hi,

I am very excited about [GitHub Actions: Environments, environment protection rules and environment secrets (beta) - GitHub Changelog](https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/)

In this open beta there is no need to sign up, all existing GitHub organizations and accounts can use the new capabilities in their public repositories and GitHub Enterprise Cloud organizations can use them in all repositories.

It is not available into my organisation repositories only in my private repositories. Would it possible to enable it on my organization ? If yes, how ?

Thanks,

**Comment by GoranHalvarsson** [comment]
Friends, may I suggest you have a look at PORT => https://docs.getport.io/
With "Self-Service" => https://docs.getport.io/create-self-service-experiences/, you can build a form that calls your github action. 
And best of all you can set permissions and approvals on it => https://docs.getport.io/create-self-service-experiences/setup-ui-for-action/#requireapproval---require-manual-approval-optional

This is not for Kubernetes only, [PORT ](https://www.getport.io/)works for any kind of projects and setups

**Comment by hp8wvvvgnj6asjm7** [comment]
can you at least show that this feature is not available for unpaid orgs?

**Reply by vdksystem** [reply]
and even for some paid orgs!!!

**Comment by drekuru** [comment]
yea this is a really annoying thing, I was going to migrate my org into github from bitbucket but this is a deal breaker. Bitbucket supports up to like 10 environments on the FREE plan. What an L

**Comment by nebuk89** [comment]
https://github.blog/changelog/2025-05-15-new-releases-for-github-actions/#actions-environments-available-in-private-repositories-for-all-plans

**Reply by ebattat** [reply]
I am still not able to see the environment in upstream organization repository. Should it be available to everyone?
